mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1114,23 +1150,28 @@ struct GridwiseGemmV2RewritePattern
     int64_t MPerBlock = tuningParams.getMPerBlock();
     int64_t NPerBlock = tuningParams.getNPerBlock();
 
-    // TODO: remove in favor of new vectorization framework
-    int64_t matrix_a_source_data_per_read =
-        op->getAttr("matrix_a_source_data_per_read")
-            .template cast<IntegerAttr>()
-            .getInt();
-    int64_t matrix_b_source_data_per_read =
-        op->getAttr("matrix_b_source_data_per_read")
-            .template cast<IntegerAttr>()
-            .getInt();
-    auto matrix_a_source_vector_read_dim = static_cast<GemmDimensions>(
-        op->getAttr("matrix_a_source_vector_read_dim")
-            .template cast<IntegerAttr>()
-            .getInt());
-    auto matrix_b_source_vector_read_dim = static_cast<GemmDimensions>(
-        op->getAttr("matrix_b_source_vector_read_dim")
-            .template cast<IntegerAttr>()
-            .getInt());
+    int64_t matrix_a_source_data_per_read = 0;
+    int64_t matrix_b_source_data_per_read = 0;
+    GemmDimension matrix_a_source_vector_read_dim;
+    GemmDimension matrix_b_source_vector_read_dim;
+
+    int64_t aCopyPerThread = (KPerBlock * KPack * MPerBlock) / blockSize;
+    int64_t bCopyPerThread = (KPerBlock * KPack * NPerBlock) / blockSize;
+    GemmDimension vectorTiebreaker =
+        (KPack > 1) ? GemmDimension::K : GemmDimension::MorN;
+    std::tie(matrix_a_source_vector_read_dim, matrix_a_source_data_per_read) =
+        bestVectorization(b, matA, aCopyPerThread, vectorTiebreaker);
+    if (matrix_a_source_vector_read_dim == GemmDimension::K) {
```

Comment:
This should be guarded by `kpack > 1`

Reply:
(because otherwise you're turning off vectorization in the k dimmension for non-kpack and that's going to kill our perf)